CONTRIBUTING.md

```diff
@@ -2,27 +2,30 @@
 
 We'd love your help! This doc covers how to become a contributor and submit code to the project.
 
+Additonal guidance can be found on [this google doc](https://docs.google.com/document/d/118a05zwIN2_Qhw7ZNmxdg2rLbatoeIcG6i-aYedzsp4/edit?usp=sharing).
+
 ## Where can I start?
 
 We tag issues that are good candidates for those new to the code with [`good first issue`](https://github.com/GoogleChrome/lighthouse/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22good+first+issue%22). We recommend you start there!
 
 ## Follow the coding style
 
-The `.eslintrc.cjs` file defines all. We use [JSDoc](http://usejsdoc.org/) with [TypeScript `checkJs`](https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html#supported-jsdoc). Annotations are encouraged for all contributions.
+The `.eslintrc.cjs` file defines all. We use [JSDoc](http://usejsdoc.org/) with [TypeScript `checkJs`](https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html#supported-jsdoc). JSDoc annotations are required for all contributions.
 
 ## Learn about the architecture
 
 See [Lighthouse Architecture](./docs/architecture.md), our overview and tour of the codebase.
 
 ## Contributing a patch
 
-If you have a contribution for our [documentation](https://web.dev/learn/#lighthouse), please submit it in the [web.dev repo](https://github.com/GoogleChrome/web.dev).
+If you have a contribution for our [documentation](https://web.dev/learn/#lighthouse), please submit it in the [web.dev repo](https://github.com/GoogleChrome/web.dev). Otherwise, for contributions to this repo:
 
-1. Submit an issue describing your proposed change.
-1. The maintainers will respond to your issue promptly.
-1. If your proposed change is accepted, and you haven't already done so, sign a Contributor License Agreement (see details below).
-1. Fork the repo, develop and test your code changes.
+1. First sign a Contributor License Agreement (see details below).
+1. If your issue has been discussed and it seems no one else is working on it, drop a note saying you plan to work take it. Feel free to ask questions. If your planned patch has no open issue already, it'd be best to first open an issue where you describe your idea and wait for us to give feedback.
+1. Fork the repo (unless you are a core contributor, then the preference is to use the main repo), develop and test your code changes.
 1. Ensure that your code adheres to the existing style in the sample to which you are contributing.
+1. We have quite a few files checked in that are derived from source files. With any change, it's good to ensure everything is up-to-date by running this command: `yarn build-report && yarn lint --fix && yarn type-check && yarn update:sample-json && yarn i18n:collect-strings`.
```

Comment:
Thanks for adding this!

Reply:
it's a moutful tho. Maybe we should add `yarn update:all`

Reply:
obv. it wouldn't include update:sample-artifacts

Reply:
Should we add `yarn update:flow-sample-json` too?

Reply:
FWIW `yarn update:sample-json` includes `yarn i18n:collect-strings` and `yarn type-check` doesn't generate anything that's checked in.

`yarn lint --fix` also seems like the odd one out here, but I guess that's debatable (but by that reasoning it seems like `yarn unit -u` or whatever would be included too)

Reply:
What checked in files does `yarn type-check` affect?

Reply:
That's the outlier, but it still seems like something that should always be checked just like the linting.

Reply:
It doesn't really fit with this item description.

I think `yarn type-check` and `yarn lint --fix` make more sense as part of the next item about testing.